### PR TITLE
RIP Fortran

### PR DESCRIPTION
**Pull request by harpolea**
This converts all remaining routines to C++ (i.e. mainly `burner_loop`).

It also converts all kernel launches to use `amrex::ParallelFor`, which means it will be much easier to debug code within lambda functions.

This currently only works with networks which have a fully C++ interface in Microphysics (e.g. `aprox13`), so should not be merged until all networks have been converted.

### CODE REVIEW

File: Exec/unit_tests/test_basestate/MaestroInit.cpp [hunk 1]
```diff
@@ -104,22 +104,6 @@ void Maestro::InitData() {
     // set p0^{-1} = p0_old
     p0_nm1.copy(p0_old);
 
-    rhoX0_old.resize(base_geom.max_radial_level + 1, base_geom.nr_fine,
-                     NumSpec);
```
Comment (zingale) [inline Exec/unit_tests/test_basestate/MaestroInit.cpp:108]: why was this removed?

Comment (harpolea) [reply Exec/unit_tests/test_basestate/MaestroInit.cpp:108]: I removed `rhoX0_old` and `rhoX0_new` entirely because they were only used by this test problem and are not needed

File: Source/MaestroBurner.cpp [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+                    aux_in[n] = s_in_arr(i, j, k, FirstAux + n);
+                }
+#endif
+                auto rhoh_in = s_in_arr(i, j, k, RhoH);
+
+                Real x_test = (ispec_threshold > 0)
+                                  ? rhoX_in[ispec_threshold] / rho_in
+                                  : 0.0;
+
+                sdc_t state_in;
```
Comment (zingale) [inline Source/MaestroBurner.cpp:286]: I don't think we have an sdc_t anymore

Comment (harpolea) [reply Source/MaestroBurner.cpp:286]: what is used now?

Comment (zingale) [reply Source/MaestroBurner.cpp:286]: it is just `burn_t`.  It has all of the fields that `sdc_t` had.

File: Source/MaestroBurner.cpp [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+                    state_in.p0 = sdc_p0;
+                    state_in.rho = rho_in;
+                    for (int n = 0; n < NumSpec; ++n) {
+                        state_in.y[n] = rhoX_in[n];
+                    }
+                    state_in.y[NumSpec] = rhoh_in;
+                    for (int n = 0; n < NumSpec; ++n) {
+                        state_in.ydot_a[n] = sdc_rhoX[n];
+                    }
+                    state_in.ydot_a[NumSpec] = sdc_rhoh;
```
Comment (zingale) [inline Source/MaestroBurner.cpp:313]: this should use the index `SENTH`

File: Source/MaestroBurner.cpp [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+                     rho_in < burning_cutoff_density_hi) &&
+                    (ispec_threshold < 0 ||
+                     (ispec_threshold > 0 &&
+                      x_test > burner_threshold_cutoff))) {
+                    state_in.p0 = sdc_p0;
+                    state_in.rho = rho_in;
+                    for (int n = 0; n < NumSpec; ++n) {
+                        state_in.y[n] = rhoX_in[n];
+                    }
+                    state_in.y[NumSpec] = rhoh_in;
```
Comment (zingale) [inline Source/MaestroBurner.cpp:309]: this should use the index `SENTH`

File: Source/MaestroBurner.cpp (hunk 3 above)
Comment (harpolea) [reply Source/MaestroBurner.cpp:313]: should the line above `state_in.y[NumSpec] = rhoh_in;` also use `SENTH`?

Comment (zingale) [reply Source/MaestroBurner.cpp]: yes

**Review by zingale — APPROVED** [review]
Everything looks good here aside from the SDC part of the burner, as noted previously
